### DISCUSSION TITLE
How to send authorization cookies when using fetch within server components?

### DISCUSSION THREAD

**Discussion by alessandrojcm** [nextauthjs/next-auth#7519] [Help]
Hi there, I am trying a POC with Next.js 13 and data fetching from a server component to a route handler. Basically, my server component tries to fetch data as described in the Next.js docs to a route handler, like so:
```typescript
async function getItems() {
  return fetch("/api/historic", {
    next: {
      tags: ["historic"],
    },
  })
    .then((r) => r.json())
    .then((r) => r.items);
}

export default async function HL() {
  const items = await getItems();
  return <HistoricList items={items} />;
}
```

My route handler is protected like so:
```typescript
export async function GET() {
  const session = await getServerSession(authOptions);
  if (!session) {
    return new NextResponse(null, {
      status: 401,
    });
  }
  // Authenticated logic
}
```
When the component tries to fetch the data, it gets a 401 from the route. If I log the cookies with `cookies()` from the route handler I can see the auth cookies not being forwarder. I tried using fetch with `credentials: 'include'` but no dice. How could I make this work?

Thanks

**Comment by gleysonabreu** [comment]
In this thread #7423 they talk about cookies not being passed. I think that in the version of next@13.4.0 or newer it already works to pass the headers.

```
import { headers } from "next/headers";
//....
async function getItems() {
  return fetch("/api/historic", {
    next: {
      tags: ["historic"],
    },
   headers: headers()
  })
    .then((r) => r.json())
    .then((r) => r.items);
}
```

**Reply by alessandrojcm** [reply]
Thanks that works! But now when I try to invalidate tags I get a ` InvalidArgumentError: invalid transfer-encoding header` when the component tries to refetch the data, don't know if this is related or if it is a different bug. Anyway I am not sure if this is the best approach for preloading/invalidating data. I wonder if I would be better off just reading the data directly from the database as this is a server component, but I would also like to invalidate just this data as opposed of invalidating the whole path.
